src/store/models/network.ts

```diff
@@ -943,6 +950,41 @@ const networkModel: NetworkModel = {
 
     actions.setAutoMineMode({ id, mode });
   }),
+  renameNode: thunk(
+    async (actions, { node, newName }, { getState, injections, getStoreActions }) => {
+      if (!newName) throw new Error(l('renameErr', { newName }));
+
+      const wasStarted = node.status === Status.Started;
+
+      if (wasStarted) {
+        await actions.stop(node.networkId);
+      }
+      const oldNodeName = node.name;
+
+      const networks = getState().networks;
+      const network = networks.find(n => n.id === node.networkId);
+      if (!network) throw new Error(l('networkByIdErr', { networkId: node.networkId }));
+
+      await renameNode(network, node, newName);
+      await getStoreActions().designer.renameNode({
+        name: newName,
+        nodeId: oldNodeName,
+      });
+
+      await injections.dockerService.renameNodeDir(network, node, newName);
```

Comment:
Ah, i see why the new name is being passed to `dockerService.renameNodeDir()`. It's because the `await renameNode(network, node, newName);` is updating the name on the `node` first. You should be able to resolve the error I mentioned above by just moving the `dockerService` rename above the `await renameNode`.